### DISCUSSION TITLE
Aceess to Android GLSurfaceView

### DISCUSSION THREAD

**Discussion by ibluesun** [dotnet/maui#3771] [General]
I want to use OpenGL ES  inside MAUI  ..  I have an already written Engine for 3D and 2D using OpenGL ES

this video shows  my progress of the app.

[![Fantastic Modeler](https://img.youtube.com/vi/6PAX3_NHzSw/0.jpg)](https://www.youtube.com/watch?v=6PAX3_NHzSw)

I used Xamarin Forms OpenTK  Control  .. which now is being deprecated.

I want to write my custom control .. maybe incorporating it inside maui if applicable   ..  where should I start    ..  and does the architecture support this.  

I know that  IOS has its Metal API   ..  Windows has DirectX12 ..  Vulkan, OpenGL ES as well in Android ..  so I am interested in exposing a control that gives the minimum surface to begin calling those APIs.

I looked around the source code but I don't have a clue where to start ..  any advices?

**Comment by orosbogdan** [comment]
Any answer on this ?
